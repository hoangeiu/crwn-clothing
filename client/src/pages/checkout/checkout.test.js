import React from "react";
import { shallow } from "enzyme";

import { CheckoutPage } from "./checkout.component";
import CheckoutItem from "../../components/checkout-item/checkout-item.component";

let wrapper;
let mockCartItems;
beforeEach(() => {
  mockCartItems = [{ id: 1 }, { id: 2 }, { id: 3 }, { id: 4 }, { id: 5 }];

  const mockProps = {
    cartItems: mockCartItems,
    total: 100,
  };

  wrapper = shallow(<CheckoutPage {...mockProps} />);
});

it("should CheckoutPage render without error", () => {
  expect(wrapper).toMatchSnapshot();
});

it("should render an equal number of CheckoutItem component as cartItems prop", () => {
  expect(wrapper.find(CheckoutItem).length).toEqual(mockCartItems.length);
});
