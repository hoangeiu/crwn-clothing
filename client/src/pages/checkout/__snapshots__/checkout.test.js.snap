// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should CheckoutPage render without error 1`] = `
<div
  className="checkout-page"
>
  <div
    className="checkout-header"
  >
    <div
      className="header-block"
    >
      <span>
        Product
      </span>
    </div>
    <div
      className="header-block"
    >
      <span>
        Description
      </span>
    </div>
    <div
      className="header-block"
    >
      <span>
        Quantity
      </span>
    </div>
    <div
      className="header-block"
    >
      <span>
        Price
      </span>
    </div>
    <div
      className="header-block"
    >
      <span>
        Remove
      </span>
    </div>
  </div>
  <Connect(CheckoutItem)
    cartItem={
      Object {
        "id": 1,
      }
    }
    key="1"
  />
  <Connect(CheckoutItem)
    cartItem={
      Object {
        "id": 2,
      }
    }
    key="2"
  />
  <Connect(CheckoutItem)
    cartItem={
      Object {
        "id": 3,
      }
    }
    key="3"
  />
  <Connect(CheckoutItem)
    cartItem={
      Object {
        "id": 4,
      }
    }
    key="4"
  />
  <Connect(CheckoutItem)
    cartItem={
      Object {
        "id": 5,
      }
    }
    key="5"
  />
  <div
    className="total"
  >
    <span>
      TOTAL: $
      100
    </span>
  </div>
  <div
    className="text-warning"
  >
    *Please use the following test credit card for payments*
    <br />
    4242 4242 4242 4242 - Exp: 01/23 CW: 123
  </div>
  <StripeCheckoutButton
    price={100}
  />
</div>
`;
